sdk/search/search-documents/src/searchServiceClient.ts

```diff
@@ -204,6 +217,29 @@ export class SearchServiceClient {
     }
   }
 
+  /**
+   * Retrieves a list of existing indexers in the service.
+   * @param options Options to the list indexers operation.
+   */
+  public async listIndexers(options: ListIndexersOptions = {}): Promise<Indexer[]> {
```

Comment:
I think we should make this generic over the select similar to `searchDocuments` on the index client so we can pick only the keys that are selected in the response shape. Otherwise `Indexer` will seem to include everything.

Reply:
Can you explain a little bit more on this? Because, Indexer does not have any nextLink parameter similar to the ones I find in searchIndexClient. So, what keys are we going to select? And the only params we get from the user is the select array. Are we trying to make that generic or something else?

Reply:
Yeah, this API is not paged as far as I know.

Basically I was thinking something like this:

```ts
interface ListIndexersOptions<Fields> {
   select?: Fields[];
}

public async listIndexers<Fields extends keyof Indexer>(options: ListIndexersOptions<Fields> = {}): Promise<Pick<Indexer, Fields>[]>
```

That way if you wrote code like

```ts
const result = await client.listIndexers({ select: ["dataSourceName", "isDisabled"]});
// after pressing dot here you'll only see dataSourceName and isDisabled in the IntelliSense menu
result[0].
```

Reply:
Thanks for the explanation. Done with the code changes and included other list ops too